src/openms/source/ANALYSIS/TARGETED/MetaboTargetedAssay.cpp

```diff
@@ -0,0 +1,498 @@
+  std::vector <MetaboTargetedAssay> MetaboTargetedAssay::extractMetaboTargetedAssay(const PeakMap& spectra,
+                                                                                    const map<BaseFeature const *, vector < size_t>>& feature_ms2_spectra_map,
+                                                                                    const double& precursor_rt_tol,
+                                                                                    const double& precursor_mz_distance,
+                                                                                    const double& cosine_sim_threshold,
+                                                                                    const double& transition_threshold,
+                                                                                    const bool& method_consensus_spectrum,
+                                                                                    const bool& exclude_ms2_precursor,
+                                                                                    const unsigned int& file_counter)
+  {
+    int transition_group_counter = 0;
+    vector <MetaboTargetedAssay> v_mta;
+
+    for (auto it = feature_ms2_spectra_map.begin();
+              it != feature_ms2_spectra_map.end();
+              ++it)
+      {
+        TargetedExperiment::Compound cmp;
+        cmp.clearMetaInfo();
+
+        vector <ReactionMonitoringTransition> v_rmt;
+
+        String description("UNKNOWN"), sumformula("UNKNOWN"), adduct("UNKNOWN");
+        StringList v_description, v_sumformula, v_adduct;
+
+        double feature_rt;
+        const BaseFeature *min_distance_feature = it->first;
+        feature_rt = min_distance_feature->getRT();
+
+        // extract metadata from featureXML
+        if (!(min_distance_feature-> getPeptideIdentifications().empty()) && !(min_distance_feature->getPeptideIdentifications()[0].getHits().empty()))
+        {
+          // accurate mass search may provide multiple possible Hits
+          // for heuristics use the identification with the smallest mz error (ppm)
+          double min_id_mz_error = std::numeric_limits<double>::max();
+          for (unsigned int j = 0; j != min_distance_feature->getPeptideIdentifications()[0].getHits().size(); ++j)
+          {
+            double current_id_mz_error = min_distance_feature->getPeptideIdentifications()[0].getHits()[j].getMetaValue("mz_error_ppm");
+            // compare the absolute error absolute error
+            if (abs(current_id_mz_error) < min_id_mz_error)
+            {
+              description = min_distance_feature->getPeptideIdentifications()[0].getHits()[j].getMetaValue("description");
+              sumformula = min_distance_feature->getPeptideIdentifications()[0].getHits()[j].getMetaValue("chemical_formula");
+              adduct = min_distance_feature->getPeptideIdentifications()[0].getHits()[j].getMetaValue("modifications");
```

Comment:
modifications is not a great name for adducts - something to keep in mind...

Reply:
Yes, this is currently the AccurateMassSearch featureXML implementation:

```
	<PeptideIdentification identification_run_ref="PI_0" score_type="" higher_score_better="true" significance_threshold="0" >
				<PeptideHit score="0" sequence="" charge="1">
					<UserParam type="stringList" name="identifier" value="[76738-62-0]"/>
					<UserParam type="stringList" name="description" value="[Paclobutrazol]"/>
					<UserParam type="string" name="modifications" value="M+H;1+"/>
					<UserParam type="string" name="chemical_formula" value="C15H20ClN3O"/>
					<UserParam type="float" name="mz_error_ppm" value="-1.604379800732327"/>
					<UserParam type="float" name="mz_error_Da" value="-4.719071027921018e-04"/>
				</PeptideHit>
```